File: citizensdk/darwin/Sources/CitizenSDK/CitizenSDKEvents.swift
import Foundation

/// Apple 平台公开的无秘密事件闭集；每项携带单调序列号供调用方保持一致观察顺序。
public enum CitizenSDKEvent: Equatable, Sendable {
    /// Invalidation only; read the latest state with the existing history API.
    case historyChanged(sequence: UInt64)
    case finalizedBlockChanged(sequence: UInt64, finalized: CitizenBlockRef)
    case lifecycleChanged(sequence: UInt64, lifecycle: CitizenSDKLifecycle)
    case capabilitiesChanged(sequence: UInt64, capabilities: CitizenSDKCapabilities)
}

File: citizensdk/android/native/src/main/kotlin/org/citizen/sdk/CitizenSdkEvents.kt
package org.citizen.sdk

/** Android 桥接层在同一有序执行器发布无秘密的历史失效、最终区块、生命周期和能力快照事件。 */
object CitizenSdkEvents {
    sealed class Event(open val sequence: String) {
        /** Invalidation only; query the existing history API for the current state. */
        class HistoryChanged(override val sequence: String) : Event(sequence)
        class FinalizedBlockChanged(
            override val sequence: String,
            val finalized: CitizenBlockRef,
        ) : Event(sequence)
        class LifecycleChanged(
            override val sequence: String,
            val lifecycle: CitizenSdkLifecycle,
        ) : Event(sequence)

        /** Carries the complete snapshot; consumers never race a follow-up query. */
        class CapabilitiesChanged(
            override val sequence: String,
            val capabilities: CitizenSdkCapabilities,
        ) : Event(sequence)

    }

    fun interface Listener {
        fun onEvent(event: Event)
    }

}
